File: app/src/main/java/za/co/riggaroo/datecountdown/db/DateTypeConverter.java
package za.co.riggaroo.datecountdown.db;


import android.arch.persistence.room.TypeConverter;

import org.threeten.bp.LocalDateTime;
import org.threeten.bp.ZoneOffset;

public class DateTypeConverter {

    @TypeConverter
    public static LocalDateTime toDate(Long timestamp) {
        return timestamp == null ? null : LocalDateTime.ofEpochSecond(timestamp, 0, ZoneOffset.ofTotalSeconds(0));
    }

    @TypeConverter
    public static Long toTimestamp(LocalDateTime date) {
        return date == null ? null : date.toInstant(ZoneOffset.ofTotalSeconds(0)).getEpochSecond();
    }
}

File: app/src/main/java/za/co/riggaroo/datecountdown/data/db/DateTypeConverter.kt
package za.co.riggaroo.datecountdown.data.db


import android.arch.persistence.room.TypeConverter

import org.threeten.bp.LocalDateTime
import org.threeten.bp.ZoneOffset

class DateTypeConverter {


    @TypeConverter
    fun toDate(timestamp: Long?): LocalDateTime? {
        return if (timestamp == null) null else LocalDateTime.ofEpochSecond(timestamp, 0, ZoneOffset.ofTotalSeconds(0))
    }


    @TypeConverter
    fun toTimestamp(date: LocalDateTime?): Long? {
        return date?.toInstant(ZoneOffset.ofTotalSeconds(0))?.epochSecond
    }
}

